File: algorithm/leetcode/48. Rotate Image/main.java
class Solution {
  private static void rotate(int[][] matrix) {
    int len = matrix.length;
    for(int i = 0;i<len;i++){
      for(int j = i;j<len;j++){
        int temp = matrix[j][i];
        matrix[j][i]=matrix[i][j];
        matrix[i][j]= temp;
      }
    }
    for(int i = 0;i<len;i++){
      for(int j = 0;j<len/2;j++){
        int temp = matrix[i][j];
        matrix[i][j]=matrix[i][len-j-1];
        matrix[i][len-j-1]= temp;
      }
    }
  }
  public static void main(String[] args) {
    int[][] tp = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}};
    rotate(tp);
    for(int i=0;i<tp.length;i++){
      for(int j=0;j<tp[i].length;j++){
        System.out.print(tp[i][j]+" ");
      }
      System.out.print(" : ");
    }
  }
}

File: algorithm/leetcode/48. Rotate Image/main.py
#!/usr/bin/env python
# coding=utf-8
'''
#Description :墨抒颖
#Author :墨抒颖
#Date :2019-11-14 21:38:36
#LastEditTime :2019-11-16 10:23:02
#LastEditors :墨抒颖
#Github :https://github.com/moshuying
#Gitee :https://gitee.com/moshuying
#Blogs :https://blog.csdn.net/qq_34846662
#Use :使用说明
'''
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        layers = n//2
        for lay in range(layers):
            for i in range(lay,n-lay-1):
                temp  = matrix[lay][i]
                matrix[lay][i] = matrix[n-1-i][lay]
                matrix[n-1-i][lay] = matrix[n-1-lay][n-1-i]
                matrix[n-1-lay][n-1-i]=matrix[i][n-1-lay]
                matrix[i][n-1-lay]=temp